src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -222,282 +259,466 @@ protected void SkipToAndParseCode(Func<SyntaxToken, bool> condition)
+                            new LocationTagged<string>(CurrentToken.Content, CurrentStart));
+                        AcceptAndMoveNext();
+                        SpanContext.EditHandler.AcceptedCharacters = AcceptedCharactersInternal.None;
+                        markupBuilder.Add(OutputAsMarkupLiteral());
+
+                        SpanContext.ChunkGenerator = SpanChunkGenerator.Null;
+                        AcceptAndMoveNext();
+                        SpanContext.EditHandler.AcceptedCharacters = AcceptedCharactersInternal.None;
+                        markupBuilder.Add(OutputAsMarkupEphemeralLiteral());
+
+                        var markupBlock = SyntaxFactory.MarkupBlock(markupBuilder.ToList());
+                        builder.Add(markupBlock);
+                    }
+                }
+                else
+                {
+                    Accept(prefixTokens);
+                    var valueStart = CurrentStart;
+                    PutCurrentBack();
+
+                    var prefix = OutputAsMarkupLiteral();
+
+                    // Dynamic value, start a new block and set the chunk generator
+                    using (var pooledResult = Pool.Allocate<RazorSyntaxNode>())
+                    {
+                        var dynamicAttributeValueBuilder = pooledResult.Builder;
+
+                        OtherParserBlock(dynamicAttributeValueBuilder);
+                        var value = SyntaxFactory.MarkupDynamicAttributeValue(prefix, SyntaxFactory.GenericBlock(dynamicAttributeValueBuilder.ToList()));
+                        builder.Add(value);
+                    }
+                }
+            }
+            else
+            {
+                Accept(prefixTokens);
+                var prefix = OutputAsMarkupLiteral();
+
+                // Literal value
+                // 'quote' should be "Unknown" if not quoted and tokens coming from the tokenizer should never have
+                // "Unknown" type.
+                var valueTokens = ReadWhile(token =>
+                    // These three conditions find separators which break the attribute value into portions
+                    token.Kind != SyntaxKind.Whitespace &&
+                    token.Kind != SyntaxKind.NewLine &&
+                    token.Kind != SyntaxKind.Transition &&
+                    // This condition checks for the end of the attribute value (it repeats some of the checks above
+                    // but for now that's ok)
+                    !IsEndOfAttributeValue(quote, token));
+                Accept(valueTokens);
+                var value = OutputAsMarkupLiteral();
+
+                var literalAttributeValue = SyntaxFactory.MarkupLiteralAttributeValue(prefix, value);
+                builder.Add(literalAttributeValue);
+            }
+        }
+
+        private void RecoverToEndOfTag(in SyntaxListBuilder<RazorSyntaxNode> builder)
```

Comment:
You'll eventually need to re-work this a little bit to instead of accepting tokens left and right without any context to be intermingling `<missing>` tokens in places that make sense.